### PR TITLE
GetKeychain now returns the last keychain transaction

### LINKED ISSUE #92: Add function to update a keychain
### Is your feature request related to a problem?

Actually, once a keychain is created, we cannot update it. We should provide function add or remove services and update the keychain.

### Describe the solution you'd like

We can create 3 functions:
- `addService` which locally add a service with it's name, derivation path, curve and hash. The 3 last parameters may have default value
- `removeService` which locally remove a service by it's name
- `update` which create a new transaction with the actual services of the local keychain object. Attention, this new transaction should also update the DID content based on the new services, and also use the same authorized public key as the previous transaction.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by bchamagne**
GetKeychain will now returns the last transaction of the keychain transaction chain (instead of the first). 
Updated the Keychain example.

Fixes #92, #93, #46

**TODO list bonus**
- [x] Add getters/setters for authorized keys (similar to services)
- [x] Move updateKeychain in Keychain class

I'm opening the PR without the above mentionned bonus, in case you want to merge it fast. I'm off for the holidays.

### CODE REVIEW

**Comment by bchamagne** [comment]
As discussed with @Neylix ,  instead of Keychain.update() as done in the previous commit 1ef33b6, it'll be cleaner to have this logic in `Account.newKeychainTransaction()`

File: lib/account.js [hunk 1]
```diff
@@ -8,31 +8,23 @@ import {
...
-        encryptedSecretKey: ecEncrypt(aesKey, key),
-      };
-    });
-
     return new this.core.transaction.builder(this.core)
       .setType("keychain")
       .setContent(JSON.stringify(keychain.toDID()))
-      .addOwnership(aesEncrypt(keychain.encode(), aesKey), authorizedKeys)
-      .build(seed, 0);
+      .addOwnership(aesEncrypt(keychain.encode(), aesKey), keychain.authorizedPublicKeys.map((key) => {
```
Comment (samuelmanzanera) [inline lib/account.js:24]: any reason for this move ?

File: lib/keychain.js [hunk 2]
```diff
@@ -140,7 +165,7 @@ export default class Keychain {
         derivationPathLength
       );
       var { byte: curveID, pos: pos } = readByte(binary, pos, 1);
-      var { byte: hashAlgoID } = readByte(binary, pos, 1);
+      var { byte: hashAlgoID, pos: pos } = readByte(binary, pos, 1);
```
Comment (samuelmanzanera) [inline lib/keychain.js:168]: why do you need this `pos` variable ?

Comment (bchamagne) [reply lib/keychain.js:168]: it is a fix when you have multiple services (cursor still need to move)

Comment (samuelmanzanera) [reply lib/keychain.js:168]: ok

File: lib/keychain.js [hunk 3]
```diff
@@ -41,6 +43,29 @@ export default class Keychain {
     return this;
   }
 
+  removeService(name) {
+    delete this.services[name];
+    return this;
+  }
+
+  // key is a Uint8array
+  addAuthorizedPublicKey(key) {
```
Comment (Neylix) [inline lib/keychain.js:52]: Here you may control the user input as in the transactionBuilder

**Review by Neylix — CHANGES_REQUESTED** [review]

**Review by Neylix — APPROVED** [review]

**Comment by Neylix** [comment]
Could you rebase and build the project to fix the conflict ?
I'm also wondering if the dist should be ignored for git as it is not really part if the code.
running the build could be done in the github action before deploying it on npm

**Comment by bchamagne** [comment]
> Could you rebase and build the project to fix the conflict ? I'm also wondering if the dist should be ignored for git as it is not really part if the code. running the build could be done in the github action before deploying it on npm

Agreed. I'll add this to the issue on current sprint

**Pull request merged**